internal/nomad/nomad.go

```diff
@@ -474,27 +477,42 @@ const (
 	UnprivilegedExecution = false
 )
 
-func prepareCommandWithoutTTY(commands []string, privilegedExecution bool) []string {
+func prepareCommandWithoutTTY(srcCommands []string, privilegedExecution bool) []string {
+	commands := make([]string, len(srcCommands)) // nozero The size is required for the copy.
+	copy(commands, srcCommands)
+
+	commands[len(commands)-1] = setInnerDebugMessages(commands[len(commands)-1])
 	commands = setUserCommand(commands, privilegedExecution)
 	commands[len(commands)-1] = fmt.Sprintf("'%s'", commands[len(commands)-1])
 	cmd := strings.Join(commands, " ")
-	return []string{"bash", "-c", fmt.Sprintf(unsetEnvironmentVariablesFormat, unsetEnvironmentVariablesShell, cmd)}
+	cmd = fmt.Sprintf(unsetEnvironmentVariablesFormat, unsetEnvironmentVariablesShell, cmd)
+	// Debug Message
+	cmd = fmt.Sprintf(timeDebugMessageFormatStart, "SetUserAndUnsetEnv", cmd)
+	return []string{"bash", "-c", cmd}
 }
 
-func prepareCommandTTY(commands []string, currentNanoTime int64, privilegedExecution bool) []string {
+func prepareCommandTTY(srcCommands []string, currentNanoTime int64, privilegedExecution bool) []string {
+	commands := make([]string, len(srcCommands)) // nozero The size is required for the copy.
+	copy(commands, srcCommands)
+
+	commands[len(commands)-1] = setInnerDebugMessages(commands[len(commands)-1])
 	commands = setUserCommand(commands, privilegedExecution)
 	// Take the last command which is the one to be executed and wrap it to redirect stderr.
 	stderrFifoPath := stderrFifo(currentNanoTime)
 	commands[len(commands)-1] =
 		fmt.Sprintf(stderrWrapperCommandFormat, stderrFifoPath, commands[len(commands)-1], stderrFifoPath)
 	cmd := strings.Join(commands, " ")
-	return []string{"bash", "-c", fmt.Sprintf(unsetEnvironmentVariablesFormat, unsetEnvironmentVariablesShell, cmd)}
+	cmd = fmt.Sprintf(unsetEnvironmentVariablesFormat, unsetEnvironmentVariablesShell, cmd)
+	// Debug Message
+	cmd = fmt.Sprintf(timeDebugMessageFormatStart, "SetUserAndUnsetEnv", cmd)
+	return []string{"bash", "-c", cmd}
 }
 
 func prepareCommandTTYStdErr(currentNanoTime int64, privilegedExecution bool) []string {
 	stderrFifoPath := stderrFifo(currentNanoTime)
-	command := []string{"bash", "-c", fmt.Sprintf(stderrFifoCommandFormat, stderrFifoPath, stderrFifoPath, stderrFifoPath)}
-	return setUserCommand(command, privilegedExecution)
+	cmd := fmt.Sprintf(stderrFifoCommandFormat, stderrFifoPath, stderrFifoPath, stderrFifoPath)
+	cmd = setInnerDebugMessages(cmd)
+	return setUserCommand([]string{"bash", "-c", cmd}, privilegedExecution)
```

Comment:
Because in contrast to the other commands, this set-user command is not wrapped in an outer `bash -c`. Since we had to deal with quite some complexity in the PR we created the other commands (/wrapped them), I did not want to open this complexity here again (also because I do not expect further findings of measuring the set-user script here again).